tests/jballerina-unit-test/src/test/resources/test-src/statements/variabledef/record-variable-definition-stmt.bal

```diff
@@ -468,10 +468,226 @@ function testRestFieldResolving() {
     testClosedRecordDefinedRestField();
 }
 
+type Topt1 record {
+    int x?;
+    int y?;
+};
+
+function testOptionalFieldRecordAssignment1() {
+    Topt1 topt = {x: 2, y: 3};
+    var {x: xx1, y: yy1} = topt;
+    assertEquality(2, xx1);
+    assertEquality(3, yy1);
+
+    topt = {y: 4};
+    var {x: xx2, y: yy2} = topt;
+    assertTrue(xx2 is ());
+    assertEquality(4, yy2);
+
+    topt = {x: 5, y: 6};
+    var {x, y} = topt;
+    assertEquality(5, x);
+    assertEquality(6, y);
+}
+
+function testOptionalFieldRecordAssignment2() {
+    Topt1 topt = {};
+    var {x, y} = topt;
+    assertTrue(x is () && y is ());
+}
+
+type Topt2 record {
+    int a;
+    record {
+        int b?;
+    }[1] c;
+};
+
+function testOptionalFieldRecordAssignment3() {
+    Topt2 topt = {a: 4, c: [{b: 5}]};
+    var {a: aa, c: cc} = topt;
+    var {b} = cc[0];
+    assertEquality(4, aa);
+    assertEquality(5, b);
+}
+
+function testOptionalFieldRecordAssignment4() {
+    Topt2 topt = {a: 4, c: [{b: 5}]};
+    var {a, c} = topt;
+    var {b} = c[0];
+    assertEquality(4, a);
+    assertEquality(5, b);
+
+    var {a: _, c: cc} = topt;
+    record {int b?;}[1] ccc = cc;
+    assertEquality(5, ccc[0].b);
+}
+
+type Topt3 record {
+    int a;
+    record {
+        int b?;
+    }[1] c?;
+};
+
+function testOptionalFieldRecordAssignment5() {
+    Topt3 topt = {a: 4, c: [{b: 5}]};
+    var {a, c} = topt;
+    assertEquality(4, a);
+    assertEquality(c is record {int b?;}[1], true);
+    if c is record {int b?;}[1] {
+        var {b} = c[0];
+        assertEquality(5, b);
+    }
+
+    topt = {a: 4};
+    var {a: _, c: cc} = topt;
+    assertTrue(cc is ());
+}
+
+type Topt4 record {
+    int x?;
+    int? y?;
+};
+
+function testOptionalFieldRecordAssignment6() {
+    Topt4 topt = {x: 2, y: 3};
+    var {x: xx, y: yy} = topt;
+    assertEquality(2, xx);
+    assertEquality(3, yy);
+}
+
+function testOptionalFieldRecordAssignment7() {
+    Topt2 topt = {a: 4, c: [{b: 5}]};
+    var {a, c: [{b}]} = topt;
+    assertEquality(4, a);
+    assertEquality(5, b);
+}
+
+function testOptionalFieldRecordAssignment8() {
+    Topt3 topt = {a: 4, c: [{b: 5}]};
+    var {a, c} = topt;
+    assertEquality(a, 4);
+    record {int b?;}[1]? _ = c;
+    var {b} = (<record {int b?;}[1]> c)[0];
+    assertEquality(b, 5);
+}
+
+function testOptionalFieldRecordAssignment9() {
+    Topt3 topt = {a: 4};
+    var {a, c} = topt;
+    assertTrue(c is ());
+}
+
+function testOptionalFieldRecordAssignment10() {
+    [record {int i?;}] y = [{}];
+    var [{i}] = y;
+    assertTrue(i is ());
+    var [{i: i2}] = y;
+    assertTrue(i2 is ());
+    y = [{i: 3}];
+    var [{i: i3}] = y;
+    assertEquality(i3, 3);
+}
+
+function testRecordDefinitionWithOptionalFields1() {
```

Comment:
This file/test is for mapping binding patterns, right? Not the correct place for these tests?